server.go

```diff
@@ -153,7 +154,18 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	ctx, span = tracer.Start(ctx, "ServeHTTP")
 	defer span.End()
 
-	s.PreRequest(w, r)
+	callbackCalled := false
+	s.AroundRequest(w, r, func() {
```

Comment:
How can we send an argument to AroundRequest that reveals the path pattern for this request? e.g. `path: "/:user/foo/:bar" so metrics/logs can be tagged accordingly?

It seems like we're pretty far away from the `configRouteEntry` from here. Would it make sense to add a `PathPattern` field to the `Route`, then only call `AroundRequest` if we have a matching route?